src/content/blog/fal.mdx

```diff
@@ -0,0 +1,99 @@
+---
+title: Val Town and Fal Partnership
+description: Bringing lightning fast AI image generation to Val Town
+pubDate: Oct 21, 2024
+author: Steve Krouse
+---
+
+import { Image } from "astro:assets";
+
+{/* https://supabase.com/blog/supabase-vercel-partnership */}
+
+[Fal](https://fal.ai/) is an API for generating AI images lightning fast.
+
+Today we're announcing a partnership with Fal to bring their API to Val Town users. You can now use Fal in your vals right away – no Fal API key necessary.
+
+## Quickstart vals
+
+The quickest way to get started is to fork one of these starter apps:
+
+{/* TODO: Side-by-side Videos or GIFs with the links underneath? */}
+
+1. [falDemoApp](https://www.val.town/v/stevekrouse/falDemoApp) - simplest possible fal app
+2. [Faltown](https://www.val.town/v/drochetti/faltown) - simplified version of Midjourney, with typesafe API client and blob storage
+
+Click fork to get a clone of this app. It'll immediately be deployed on your account with no further setup necessary. You can then ask Townie to customize it to your needs. Want a DnD character generator? [No problem!](https://stevekrouse-genuinevioletswordtail.web.val.run/) Want a tool to help you design your next tattoo? [Done!](https://stevekrouse-tattoodesignapp.web.val.run/)
+
+## Instructions for Townie
+
+Another quick way to get started is to copy & paste the following instructions into Townie:
+
+````markdown
+Import the necessary modules:
+
+```ts
+import { falProxyRequest } from "https://esm.town/v/stevekrouse/falProxyRequest";
+import { createFalClient } from "https://esm.sh/@fal-ai/client";
+```
+
+Set up the Fal client on the client side:
+
+```ts
+const FAL_ENDPOINT = "fal-ai/flux/schnell";
+const fal = createFalClient({
+  proxyUrl: "/api/fal/proxy",
+});
+```
+
+Use the Fal client to handle image generation requests on the client:
+
+```ts
+const result = await fal.run("fal-ai/flux/schnell", {
+  input: {
+    prompt,
+    image_size: "landscape_4_3",
+    num_inference_steps: 4, // must be 12 or less
+    num_images: 1,
+    enable_safety_checker: true,
+    sync_mode: true,
+  },
+});
+const image_url = result.data.images[0].url;
+```
+
+Set up the Fal proxy request on the server side:
+
+```ts
+export default async function server(req: Request): Promise<Response> {
+  const url = new URL(req.url);
+  if (url.pathname === "/") {
+    /* Return HTML */
+  }
+
+  if (url.pathname === "/api/fal/proxy") {
+    return falProxyRequest(req);
+  }
+}
+```
+````
+
```

Comment:
small suggestion: we could briefly talk about the Flux Schnell, the fast image model being used here and link `https://fal.ai/flux` for more info.

Reply:
done!